File: src/app/Foundation.Extension.Admin-UI/domain/models/dispositions/column.ts
export class Column {
    id: string;
    value: string;
    index: number;
    hidden: boolean;
    sortable: boolean;
    filterable: boolean;
    configurable: boolean;
    label: string;
    disabled: boolean;

    constructor(params: ColumnDTO) {
        this.id = params.id;
        this.value = params.value;
        this.index = params.index;
        this.hidden = params.hidden;
        this.sortable = params.sortable;
        this.filterable = params.filterable;
        this.configurable = params.configurable;
        this.label = params.label;
        this.disabled = params.disabled;
    }
}

export interface ColumnDTO {
    id: string;
    value: string;
    index: number;
    hidden: boolean;
    sortable: boolean;
    filterable: boolean;
    configurable: boolean;
    label: string;
    disabled: boolean;
}

File: src/shared/Foundation.Extension.Domain/Models/Dispositions/Column.cs
using System;
using System.Collections.Generic;
using Foundation.Extension.Domain.Enums;

namespace Foundation.Extension.Domain.Models
{
    public class Column
    {
        public Guid Id { get; set; }

        public Guid TableId { get; set; }
        public string TableCode { get; set; }

        public PropertyType PropertyType { get; set; }
        public Guid? EntityPropertyId { get; set; }
        public Guid? CustomPropertyId { get; set; }

        public string Code { get; set; }
        public string Value { get; set; }
        
        public string Label { get; set; }

        public bool Sortable { get; set; }
        public bool Filterable { get; set; }

        public int Index { get; set; }
        public bool Hidden { get; set; }
        public bool Disabled { get; set; }
    }
}